### ISSUE TITLE
add strategy="mode" for fill_null

### ISSUE THREAD

**Issue by zippeurfou** [pola-rs/polars#16585]
### Description

This is a small feature request to add the strategy "mode" for fill_null. This is especially useful for column that are string.
Doing it via code is straighforward. 
If not a feature request maybe the doc could provide an example with this instead of using it the median.

**Comment by s-banach** [comment]
What if there is not a unique mode?

**Comment by zippeurfou** [comment]
This is a good point. https://docs.pola.rs/py-polars/html/reference/series/api/polars.Series.mode.html do mention the possibility of having multiples values. I guess this would involve having a method to tie break them which could be none or random for example. This will then be a feature request for mode.

**Comment by ritchie46** [comment]
You can do that with a `when -> then -> otherwise`:

```python
foo = pl.col("foo")
pl.when(foo.is_null()).then(foo.mode().first())

```

**Comment by zippeurfou** [comment]
Thanks @ritchie46, I was not mentioning it is not possible. I actually use the `fill_null` that you mentioned in your example but I was referring that it could be good to either add this example in the doc or as a `strategy` solution for the method arguments in https://github.com/pola-rs/polars/blob/py-0.20.30/py-polars/polars/dataframe/frame.py#L7202 as this is pretty helpful for string.

**Comment by coastalwhite** [comment]
I don't think this is really needed. You could also compose it from `expr.fill_null(expr.mode().implode())`.
